src/mgmt/srv_pool.c

```diff
@@ -573,47 +574,66 @@ int
 ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force)
 {
 	struct mgmt_svc	*svc;
-	int		rc;
+	d_rank_list_t	*psvcranks;
+	int		 rc;
 
-	/* TODO check metadata about the pool's existence?
-	 *      and check active pool connection for "force"
-	 */
 	D_DEBUG(DB_MGMT, "Destroying pool "DF_UUID"\n", DP_UUID(pool_uuid));
 
 	rc = ds_mgmt_svc_lookup_leader(&svc, NULL /* hint */);
 	if (rc != 0)
 		goto out;
 
+	/* Get pool service ranks before pool_destroy_prepare() that changes
+	 * pr_state to destroying.
+	 */
+	rc = pool_get_svc_ranks(svc, pool_uuid, &psvcranks);
+	if (rc != 0) {
+		D_ERROR("Failed to get pool service ranks "DF_UUID" rc: %d\n",
+			DP_UUID(pool_uuid), rc);
+		goto out_svc;
+	}
+
+	/* Check active pool connections, evict only if force */
+	rc = ds_pool_svc_check_evict(pool_uuid, psvcranks, force);
+	if (rc != 0) {
+		D_ERROR("Failed to check/evict pool handles "DF_UUID" rc: %d\n",
+			DP_UUID(pool_uuid), rc);
+		goto out_ranks;
+	}
+
 	rc = pool_destroy_prepare(svc, pool_uuid);
 	if (rc != 0) {
-		if (rc == -DER_ALREADY)
+		if (rc == -DER_ALREADY) {
 			rc = 0;
-		goto out_svc;
+			goto out_ranks;
+		}
```

Comment:
If rc != 0 && rc != -DER_ALREADY, we should also goto out_ranks, no?

Reply:
I restored this original logic in the updated patch.